src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -6128,8 +6194,23 @@ private static object PostOneBeginConnect(MultipleAddressConnectAsyncResult cont
                 // MSRC 11081 - Do the necessary security demand
                 context.socket.CheckCacheRemote(ref endPoint, true);
 
-                IAsyncResult connectResult = context.socket.UnsafeBeginConnect(endPoint,
-                    new AsyncCallback(MultipleAddressConnectCallback), context);
+                var asyncCallback = new AsyncCallback(MultipleAddressConnectCallback);
```

Comment:
Any reason not to cache this delegate in a static rather than allocating each time?